### ISSUE TITLE
Storing java.math.BigDecimal silently "disconnects" from storage

### ISSUE THREAD

**Issue by pm4r** [replikativ/datahike#5]
Example:
```clojure
(def uri (str "datahike:file://" (.getAbsolutePath (java.io.File. "testdb"))))
(d/create-database uri)
(def conn (d/connect uri))

@(d/transact conn [{:my/prop1 42.5}])
(d/seek-datoms @conn :eavt) ; all fine, our single datom is there

@(d/transact conn [{:my/prop2 42.5M}])
(d/seek-datoms @conn :eavt) ; still fine, we he two datoms now
(d/q '[:find [(pull ?e [*]) ...]
       :where [?e :my/prop2]]
  @conn)
(type (:my/prop2 (first *1))) ; -> java.math.BigDecimal

@(d/transact conn [{:my/prop3 "abc"}])
(d/seek-datoms @conn :eavt) ; now we have three datoms

(d/release conn)

(def conn (d/connect uri))
(d/seek-datoms @conn :eavt) ; only our first datom is there

(d/release conn)
(d/delete-database uri)
```

It doesn't matter if db was created with or without schema. Until disconnection we have all transacted data available but after reconnection everything since inserting bigdec is not available.

I would expect inserting unsupported data type should throw exception.

**Comment by whilo** [comment]
This issue should be fixed with the release of `0.1.1`. Can you please verify? 

P.S.: What are you working on btw.? I have recently needed BigDecimal support as well to deal with accounting.

**Comment by pm4r** [comment]
I works now for bigdec, thanks :). Although it came out I don't need them at all at the moment. 

I'm working on mobile app (written entirely in clojurescript, using Expo :D) which needs backend for image recognition and storing some user accounts and orders. It will be low volume, special purpose app, so I gave datahike a shot to simplify my server deployment. I consider it as pretty safe shot, because I can always fall back to datomic free if something will crash to often ;). But for now it looks like it fits my little needs, and I have plans to use it on my next project.
